### DISCUSSION TITLE
Which ports to forward

### DISCUSSION THREAD

**Discussion by Spiritreader** [qe201020335/BeatTogether-DockerCompose#8] [Q&A] [answered]
My firewall rules are drop by default, so I need to figure out all ports that need to be forwarded.

If I understand this correctly, the ports that the game needs are:

- base ports `30000` plus the number of defined lobbies. So for 3 lobbies that would need to be `30000-30002`
- GameLift port `8989`
- Status API port `23280`

Is the masterserver endpoint also required at `2328`?

Are the ports UDP or TCP?

**Comment by qe201020335** [comment, marked as answer]
> base ports 30000 plus the number of defined lobbies. So for 3 lobbies that would need to be 30000-30002 

This is correct, but I suggest you also open a couple more ports, in case it didn't reuse an old lobby port in time. This needs to be UDP

Status API is TCP. It is a rest API.

The master server `2328` is for backwards compatibility and is UDP
GameLift `8989` is TCP 

Here is a quick screenshot of `netstat`
![image](https://github.com/qe201020335/BeatTogether-DockerCompose/assets/31044781/0ad1d397-d325-4bb7-8078-1cbea844f731)

**Reply by Spiritreader** [reply]
Thank you! It works :)

One thing that I did notice is with the configuration jsons.
I could not get rabbitmq to pick up properly in my own custom docker compose network (not using host essentially, and the correct configuration to set the rabbitmq url was this:

```
  "RabbitMQ": {
    "HostName": "rabbitmq"
  },
```

in the dedicated.json and master.json
